DeployClient/Program.cs

```diff
@@ -178,9 +199,17 @@ static void Main(string[] args)
                             break;
                         }
 
-                        // Sleep.
                         System.Threading.Thread.Sleep(interval);
-                    }
+
+                        var success = API.GetSession(sessionGuid, out response);
+
+                        // The api should not be returning a 404 status at this point
+                        if (!success)
+                        {
+                            throw new Exception("Remote API returned status 404");
```

Comment:
Please can we throw something more specific than `System.Exception`. It's [bad practise](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types#exception-and-systemexception) to throw these. I think `System.Web.HttpException` would be more appropriate instead.